---
title: "FGF- and FGFR-coding gene alterations in urothelial cancers"
subtitle: "Short analysis report"
author: "Piotr Tymoszuk for the Department of Urology, Medical University of Innsbruck"
date: "`r format(Sys.time(), '%Y-%m-%d')`"

output: 
  bookdown::html_document2:
    css: "style.css"
    
bibliography: blca.bib

csl: frontiers_medical.csl

header-includes:
  \usepackage{longtable}
  \usepackage{tabu}
  \usepackage{caption}
  \usepackage{makecell}
  \usepackage{pdflscape}
  \usepackage{array}
  \usepackage{booktabs}
  \usepackage{threeparttable}
  \usepackage{threeparttablex}
  \usepackage{wrapfig}
  \usepackage{multirow}
  \usepackage[normalem]{ulem}
  \usepackage{colortbl}
  \usepackage{xcolor}
  \usepackage{float} \floatplacement{figure}{H} \floatplacement{table}{H}

---

```{r, setup, include = FALSE}

library(Cairo)

opts_chunk$set(echo = FALSE, 
                      warning = FALSE, 
                      message = FALSE, 
                      dev = "CairoPNG", 
                      dpi = 600)

set_flextable_defaults(font.family = "Cambria", 
                       font.size = 10)


```

\newpage

# Results

## Characteristic of investigated cohorts and frequency of genetic alterations

```{r fig-general-freq, fig.width = figur::convert(rep_figs$general_freq, to = 'in')$w, fig.height = figur::convert(rep_figs$general_freq, to = 'in')$h, fig.cap = 'The most frequent somatic mutations and gene copy alterations in urothelial cancer.'}

rep_figs$general_freq$plot

```

__Figure \@ref(fig:fig-general-freq). The most frequent somatic mutations and gene copy alterations in urothelial cancer.__ 

_Percentages of somatic mutations, gene deletions and amplification present in at least 5% of cancers samples in all investigated cohorts (`r globals$cohort_labs %>% paste(collapse = ', ')`, note: gene amplification and deletion data were not available for the `r globals$cohort_labs["imvigor"]` cohort) are presented in bar plots. Alterations affecting FGF- and FGFR-coding genes are highlighted with bold font. Numbers of analyzed cancer samples are indicated in the plot legend._

\newpage

```{r fig-fgfr-freq, fig.width = figur::convert(rep_figs$fgfr_freq, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_freq, to = 'in')$h, fig.cap = 'Frequency of somatic mutations and copy number alterations of FGF- and FGFR-coding genes.'}

rep_figs$fgfr_freq$plot

```

__Figure \@ref(fig:fig-fgfr-freq). Frequency of somatic mutations and copy number alterations of FGF- and FGFR-coding genes.__ 

_Percentages of somatic mutations, deletions and amplifications of FGF- and FGFR-coding genes in the investigated cohort are presented in bar plots. Note that amplification and deletion data were not available for the `r globals$cohort_labs["imvigor"]` cohort. Numbers of analyzed cancer samples are indicated in the plot legend._

\newpage

```{r tab-cohorts, tab.cap = 'Characteristic of the investigated cohorts. Numeric features are presented as medians with interquartile ranges and ranges. Qualitative variables are shown as percentages and observation counts of the categories.'}

flextable::flextable(rep_tabs$cohorts) %>% 
  set_widths(c(3, rep(3.5, 3), 2.8, 2.8)) %>% 
  footnote(1, 5:6, 
           value = as_paragraph('Numeric variables: Kruskal-Wallis test with \u03B7\u00B2 effect size statistic. Categorical variables: \u03C7\u00B2 test with Cramer V effect size statistic.'), 
           part = 'header', 
           ref_symbols = 'a') %>% 
  theme_vanilla

```

\newpage

```{r tab-genetics, tab.cap = "Frequency of the most common somatic mutations and copy number variants in the investigated cohorts. Frequencies of alterations present in at least 5% of cancer samples are displayed. Statistical significance for differences between the cohorts was assessed by chi-squared test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file."}

rep_tabs$genetics[1, 1] %>% 
  mutate(`Alteration type` = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

```{r tab-fgf-genetics, tab.cap = "Frequency of somatic mutations and copy number variants of FGF- and FGFR-coding genes in the investigated cohorts. Statistical significance for differences between the cohorts was assessed by chi-squared test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file."}

rep_tabs$fgf_genetics[1, 1] %>% 
  mutate(`Alteration type` = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

## Classification of mutations in FGFR-coding genes, and their possible effects on protein structure and function

```{r fig-mutation-type, fig.width = figur::convert(rep_figs$mutation_type, to = 'in')$w, fig.height = figur::convert(rep_figs$mutation_type, to = 'in')$h, fig.cap = 'Frequency of somatic mutations in FGFR1/2/3/4 genes split by mutation type.'}

rep_figs$mutation_type$plot

```

__Figure \@ref(fig:fig-mutation-type). Frequency of somatic mutations in FGFR1/2/3/4 genes split by mutation type.__ 

_Mutations in FGFR1, FGFR2, FGFR3, and FGFR4 genes in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` cohorts were classified by their effects on the coding sequence. Their frequency was expressed as percentage of cancer samples and presented in stack plots. Numbers of analyzed cancer samples are displayed in the plot captions. _

\newpage

```{r fig-variant-type, fig.width = figur::convert(rep_figs$variant_type, to = 'in')$w, fig.height = figur::convert(rep_figs$variant_type, to = 'in')$h, fig.cap = 'Frequency of somatic mutations in FGFR1/2/3/4 genes split by variant type.'}

rep_figs$variant_type$plot

```

__Figure \@ref(fig:fig-variant-type). Frequency of somatic mutations in FGFR1/2/3/4 genes split by variant type.__ 

_Mutations in FGFR1, FGFR2, FGFR3, and FGFR4 genes in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` cohorts were classified by variant type. Their frequency was expressed as percentage of cancer samples and presented in stack plots. Numbers of analyzed cancer samples are displayed in the plot captions. SNP: single nucleotide polymorphism, DNP: di-nucleotide polymorphism;  ONP: oligo-nucleotide polymorphism._

\newpage

```{r fig-protein-domain-split, fig.width = figur::convert(rep_figs$protein_domain_split, to = 'in')$w, fig.height = figur::convert(rep_figs$protein_domain_split, to = 'in')$h, fig.cap = 'Frequency of somatic mutations in FGFR1/2/3/4 genes split by protein domain.'}

rep_figs$protein_domain_split$plot

```

__Figure \@ref(fig:fig-protein-domain-split). Frequency of somatic mutations in FGFR1/2/3/4 genes split by protein domain.__ 

_Mutations in FGFR1, FGFR2, FGFR3, and FGFR4 genes in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` cohorts were classified by protein domain. Their frequency was expressed as percentage of cancer samples and presented in stack plots. Numbers of analyzed cancer samples are displayed in the plot captions._

\newpage

```{r tab-fgfr-class, tab.cap = 'Frequency of mutations in the FGFR1/2/3/4 genes in the GENIE BLCA and TCGA BLCA cohorts classified by mutation type, variant type, and protein domain. The table is available as a supplementary Excel file.'}

rep_tabs$fgfr_class[1, 1] %>% 
  mutate(`Classification scheme` = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r fig-fgfr1-position, fig.width = figur::convert(rep_figs$fgfr1_position, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr1_position, to = 'in')$h, fig.cap = 'Positions of FGFR1 protein affected by somatic mutations.'}

rep_figs$fgfr1_position$plot

```

__Figure \@ref(fig:fig-fgfr1-position). Positions of FGFR1 protein affected by somatic mutations.__ 

_Somatic mutations in the FGFR1 gene in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` were classified by their effects on nucleotide sequence and mapped to amino acids of the FGFR1 protein. Mutation localization and type are visualized in scatter plots below the FGFR1 protein domain schemes. Each point represents a single somatic mutation. Point color codes for protein domain. Total mutation numbers are displayed in the plot captions._

\newpage

```{r fig-fgfr2-position, fig.width = figur::convert(rep_figs$fgfr2_position, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr2_position, to = 'in')$h, fig.cap = 'Positions of FGFR2 protein affected by somatic mutations.'}

rep_figs$fgfr2_position$plot

```

__Figure \@ref(fig:fig-fgfr2-position). Positions of FGFR2 protein affected by somatic mutations.__ 

_Somatic mutations in the FGFR2 gene in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` were classified by their effects on nucleotide sequence and mapped to amino acids of the FGFR2 protein. Mutation localization and type are visualized in scatter plots below the FGFR2 protein domain schemes. Each point represents a single somatic mutation. Point color codes for protein domain. Total mutation numbers are displayed in the plot captions._

\newpage

```{r fig-fgfr3-position, fig.width = figur::convert(rep_figs$fgfr3_position, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr3_position, to = 'in')$h, fig.cap = 'Positions of FGFR3 protein affected by somatic mutations.'}

rep_figs$fgfr3_position$plot

```

__Figure \@ref(fig:fig-fgfr3-position). Positions of FGFR3 protein affected by somatic mutations.__ 

_Somatic mutations in the FGFR3 gene in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` were classified by their effects on nucleotide sequence and mapped to amino acids of the FGFR3 protein. Mutation localization and type are visualized in scatter plots below the FGFR3 protein domain schemes. Each point represents a single somatic mutation. Point color codes for protein domain. Total mutation numbers are displayed in the plot captions._

\newpage

```{r fig-fgfr4-position, fig.width = figur::convert(rep_figs$fgfr4_position, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr4_position, to = 'in')$h, fig.cap = 'Positions of FGFR4 protein affected by somatic mutations.'}

rep_figs$fgfr4_position$plot

```

__Figure \@ref(fig:fig-fgfr4-position). Positions of FGFR4 protein affected by somatic mutations.__ 

_Somatic mutations in the FGFR4 gene in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` were classified by their effects on nucleotide sequence and mapped to amino acids of the FGFR4 protein. Mutation localization and type are visualized in scatter plots below the FGFR4 protein domain schemes. Each point represents a single somatic mutation. Point color codes for protein domain. Total mutation numbers are displayed in the plot captions._

\newpage

## Co-occurrence of genetic alterations in FGF- and FGFR-coding genes with the most frequent genetic features of urothelial cancers

```{r fig-genetic-overlap, fig.width = figur::convert(rep_figs$genetic_overlap, to = 'in')$w, fig.height = figur::convert(rep_figs$genetic_overlap, to = 'in')$h, fig.cap = 'Co-occurrence of the most common genetic alterations in urothelial cancer.'}

rep_figs$genetic_overlap$plot

```

__Figure \@ref(fig:fig-genetic-overlap). Co-occurrence of the most common genetic alterations in urothelial cancer.__ 

_Co-occurrence of the most common genetic alterations in urothelial cancers (at least 5% of cancer samples) was investigated in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` cohorts by pairwise Jaccard's similarity coefficients and two-dimensional MDS (multi-dimensional scaling) of pairwise Jaccard's distances._

_(A) MDS results are visualized in scatter plots with single genetic alterations depicted as points. Point color codes for alteration type (somatic mutation, deletion, amplification), point size represents frequency of the alteration in the data set. Note that genetic features displayed in the plots close to each other are expected to co-occur in a high fraction of cancer samples. Total numbers of samples are indicated in the plot captions._

_(B) Frequencies of samples with and without genetic alterations are visualized as heat maps of contingency tables for selected pairs of genetic features. Sample percentages and counts within the total sample sets are indicated in the heat map tiles. Jaccard's similarity coefficients and with their false discovery rate-corrected significance ($pFDR(J > 0)$) are presented in the plot captions._

\newpage

```{r fig-fgfr-overlap, fig.width = figur::convert(rep_figs$fgfr_overlap, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_overlap, to = 'in')$h, fig.cap = 'Co-occurrence of genetic alterations of FGF and FGFR genes in urothelial cancer.'}

rep_figs$fgfr_overlap$plot

```

__Figure \@ref(fig:fig-fgfr-overlap). Co-occurrence of genetic alterations of FGF and FGFR genes in urothelial cancer.__ 

_Co-occurrence of somatic mutations, deletions, and amplifications of FGF- and FGFR-coding genes was investigated in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` cohorts by pairwise Jaccard's similarity coefficients and two-dimensional MDS (multi-dimensional scaling) of pairwise Jaccard's distances._

_(A) MDS results are visualized in scatter plots with single genetic alterations depicted as points. Point color codes for alteration type (somatic mutation, deletion, amplification), point size represents frequency of the alteration in the data set. Note that genetic features displayed in the plots close to each other are expected to co-occur in a high fraction of cancer samples. Total numbers of samples are indicated in the plot captions._

_(B) Frequencies of samples with and without genetic alterations are visualized as heat maps of contingency tables for selected pairs of genetic features. Sample percentages and counts within the total sample sets are indicated in the heat map tiles. Jaccard's similarity coefficients and with their false discovery rate-corrected significance ($pFDR(J > 0)$) are presented in the plot captions._

\newpage

## Clinical and biological significance of genetic alterations of FGF- and FGFR-coding genes

```{r tab-fgfr3-clinic, tab.cap = 'Demographic, clinical, and pathological characteristic of cancers with WT and mutated FGFR3. Numeric features are presented as medians with interquartile ranges and ranges. Qualitative variables are shown as percentages and observation counts of the categories.'}

flextable::flextable(rep_tabs$fgfr3_clinic) %>% 
  set_widths(c(2.2, 3, rep(3.5, 2), 2.8, 2.8)) %>% 
  footnote(1, 5:6, 
           value = as_paragraph('Numeric variables: Kruskal-Wallis test with \u03B7\u00B2 effect size statistic. Categorical variables: \u03C7\u00B2 test with Cramer V effect size statistic.'), 
           part = 'header', 
           ref_symbols = 'a') %>% 
  theme_vanilla

```

\newpage

```{r fig-fgfr-correlation, fig.width = figur::convert(rep_figs$fgfr_correlation, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_correlation, to = 'in')$h, fig.cap = 'Correlation of mRNA levels of FGF- and FGFR-coding genes.'}

rep_figs$fgfr_correlation$plot

```

__Figure \@ref(fig:fig-fgfr-correlation). Correlation of mRNA levels of FGF- and FGFR-coding genes.__ 

_Pairwise correlation of log2-transformed mRNA levels of FGF- and FGFR-coding genes was investigated in the `r globals$cohort_labs[c("tcga", "imvigor")] %>%  paste(collapse = ' and ')` cohorts by false discovery rate-adjusted Pearson's test. Correlation coefficients r for significant gene pairs are depicted in bubble plots. Color of the points codes for correlation sign. Point size represents absolute values of r. Numbers of analyzed cancer samples are displayed in the plot captions._

\newpage

```{r fig-fgfr-dge, fig.width = figur::convert(rep_figs$fgfr_dge, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_dge, to = 'in')$h, fig.cap = 'Expression of FGF- and FGFR-coding genes in urothelial cancers stratified by presence of FGFR3 mutations.'}

rep_figs$fgfr_dge$plot

```

__Figure \@ref(fig:fig-fgfr-dge). Expression of FGF- and FGFR-coding genes in urothelial cancers stratified by presence of FGFR3 mutations.__ 

_Differences in expression of FGF- and FGFR-coding genes between cancers with and without somatic mutations in the FGFR3 gene were assessed by two-tailed T test with Cohen's d effect size metric. P values were corrected with the false discovery rate (FDR) method. Differentially regulated genes were identified by pFDR < 0.05 and at least 1.5 fold-regulation of gene expression in mutated samples as compared with WT specimens._

_(A) pFDR values and log2 fold-regulation estimates of gene expression in mutated vs WT samples are presented in Volcano plots. Each point represents a single gene. Point color codes for significance and regulation sign, differentially regulated genes are labeled with their symbols. The significance and fold-regulation cutoffs of differential gene expression are depicted as dashed lines. Numbers of analyzed cancer samples are indicated in the plot caption._

_(B) log2-transformed mRNA levels for genes found to be differentially regulated in both cohorts are presented in box plots. Median expression values with interquartile ranges are depicted as boxes with whiskers spanning over 150% of the interquartile ranges. Single observations are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of analyzed cancer samples are indicated in the X axes._

\newpage

```{r tab-fgfr3-dge, tab.cap = "Differential expression of FGF- and FGFR-coding genes in urothelial cancers with and without FGFR3 mutations. log2-transformed expression levels are presented as medians with interquartile ranges and ranges. Statistical significance was determined by two-tailed T test with Cohen's d effect size statistic. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file."}

rep_tabs$fgfr3_dge[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r fig-fgfr-tmb, fig.width = figur::convert(rep_figs$fgfr_tmb, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_tmb, to = 'in')$h, fig.cap = 'Total mutation burden and counts of copy number alterations in FGFR3 WT and FGFR3-mutated cancers.'}

rep_figs$fgfr_tmb$plot

```

__Figure \@ref(fig:fig-fgfr-tmb). Total mutation burden and counts of copy number alterations in FGFR3 WT and FGFR3-mutated cancers.__ 

_Total mutation burdens (TMB) were provided for the cohorts by the study authors. Counts of mutations, deletions and amplifications were computed for each cancer sample based on the published information. Note, that information on deletions and amplification was not available for the `r globals$cohort_labs["imvigor"]` cohort. TMB, and counts of mutations, deletions and amplifications were compared between cancers with and without FGFR3 mutations by Mann-Whitney test with r effect size statistic. P values were adjusted for multiple testing with the false discovery rate method. Median values with interquartile ranges are shown as boxes with whiskers spanning over 150% of the interquartile ranges. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of analyzed cancer samples are indicated in the X axes._

\newpage

```{r fig-fgfr-os, fig.width = figur::convert(rep_figs$fgfr_os, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_os, to = 'in')$h, fig.cap = 'Overall survival of patients with FGFR3 WT and FGFR3-mutated cancers.'}

rep_figs$fgfr_os$plot

```

__Figure \@ref(fig:fig-fgfr-os). Overall survival of patients with FGFR3 WT and FGFR3-mutated cancers.__ 

_Overall survival information was provided for the `r globals$cohort_labs["tcga"]` study participants. Approximate overall survival was calculated for the `r globals$cohort_labs["genie"]` cohort based on age at enrollment in years and age at the last follow-up or death in days. Overall survival was subsequently compared between patients with cancers with and without FGFR3 mutations. Statistical significance was determined by false discovery rate-adjusted Peto-Peto test. Fractions of surviving patients are presented in Kaplan-Meier plots. P values are indicated in the plots, numbers of observations and deaths are displayed in the plot captions. Numbers of observations and deaths in the FGFR3 mutation strata are presented in the plot legends._

\newpage

```{r fig-fgfr-rfs, fig.width = figur::convert(rep_figs$fgfr_rfs, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_rfs, to = 'in')$h, fig.cap = 'Disease-specific and relapse-free survival of patients with FGFR3 WT and FGFR3 mutated cancers.'}

rep_figs$fgfr_rfs$plot

```

__Figure \@ref(fig:fig-fgfr-rfs). Disease-specific and relapse-free survival of patients with FGFR3 WT and FGFR3 mutated cancers.__ 

_Disease-specific and relapse-free survival information was provided for the `r globals$cohort_labs["tcga"]` cohort. Differences in survival between patients with cancers with and without FGFR3 mutations were investigated by false discovery rate-adjusted Peto-Peto test. Fractions of surviving patients are presented in Kaplan-Meier plots. P values are indicated in the plots, numbers of observations and deaths are displayed in the plot captions. Numbers of observations and deaths in the FGFR3 mutation strata are presented in the plot legends._ 

\newpage

```{r fig-amp11q13, fig.width = figur::convert(rep_figs$amp11q13, to = 'in')$w, fig.height = figur::convert(rep_figs$amp11q13, to = 'in')$h, fig.cap = 'Expression of FGF- and FGFR-coding genes in cancers with and without amplification of the 11q13 chromosome region.'}

rep_figs$amp11q13$plot

```

__Figure \@ref(fig:fig-amp11q13). Expression of FGF- and FGFR-coding genes in cancers with and without amplification of the 11q13 chromosome region.__ 

_Amplification of the 11q13 chromosome region with FGF3, FGF4, FGF19 and CCND genes was identified as frequent genetic alteration in urothelial carcinoma. Differences in expression of FGF- and FGFR-coding genes between cancers with and without 11q13 amplification were assessed in the `r globals$cohort_labs["tcga"]` cohort by Mann-Whitney test with r effect size metric. P values were corrected for multiple testing with the false discovery rate method. Median log2-transformed mRNA levels with interquartile ranges are represented by boxes with whiskers spanning over 150% of the interquartile ranges for the significantly regulated genes. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of analyzed cancer samples are indicated in the X axes._

\newpage

## Consensus molecular subtypes of urothelial cancers

```{r tab-cons-genetics, tab.cap = "Frequency of somatic mutations and copy number variants of FGF- and FGFR-coding genes in the consensus molecular classes of urothelial cancers. The frequencies are presented as percentages of the consensus class. Statistical significance of differences between the consensus classes was determined by chi-squared test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file."}

rep_tabs$cons_genetics[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r fig-fgfr-subtypes, fig.width = figur::convert(rep_figs$fgfr_subtypes, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_subtypes, to = 'in')$h, fig.cap = 'Alterations of FGF- and FGFR-coding genes in consensus molecular classes of urothelial cancers.'}

rep_figs$fgfr_subtypes$plot

```

__Figure \@ref(fig:fig-fgfr-subtypes). Alterations of FGF- and FGFR-coding genes in consensus molecular classes of urothelial cancers.__ 

_Cancer samples from the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` data sets were assigned to the MIBC consensus molecular subtypes based on ComBat-adjusted whole-genome mRNA levels by the nearest centroid algorithm (R package consensusMIBC)._ 
_Differences in frequency of somatic mutations and copy number variants of FGF- and FGFR-coding genes between the consensus molecular classes were assessed by $\chi^2$ test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Note that copy number information was not available for the `r globals$cohort_labs["imvigor"]` cohort._

_(A) Percentages of cancer samples in the consensus molecular classes were visualized in a stack plot. Statistical significance of difference in class distribution between the cohorts was assessed by $\chi^2$ test with Cramer's V effect size statistic. The effect size and p value are shown in the plot caption. Total numbers of cancer samples are indicated in the Y axis._

_(B) Presence/absence of selected genetic alterations in the consensus molecular classes visualized in heat maps. Alteration names with effect sizes and p values of differences between the consensus classes are indicated in the Y axis. Significant effects are highlighted with bold font. Numbers of cancer samples in the consensus classes are displayed in the plot captions._

_LumP: luminal papillary; LumU: luminal genetically unstable; LumNS: luminal non-specified; Stroma-rich: luminal stroma-rich; Ba/Sq: basal/squamous-like; NE-like: neuroendocrine-like._

\newpage

```{r fig-fgfr-subdetails, fig.width = figur::convert(rep_figs$fgfr_subdetails, to = 'in')$w, fig.height = figur::convert(rep_figs$fgfr_subdetails, to = 'in')$h, fig.cap = 'Frequency of somatic mutations of FGFFR3 and amplification of FGF3/4/19 in consensus molecular classes of urothelial cancer.'}

rep_figs$fgfr_subdetails$plot

```

__Figure \@ref(fig:fig-fgfr-subdetails). Frequency of somatic mutations of FGFFR3 and amplification of FGF3/4/19 in consensus molecular classes of urothelial cancer.__ 

_Assignment of cancer samples to consensus molecular classes and testing for differences in genetic alterations between the consensus classes were presented in Figure \@ref(fig:fig-fgfr-subtypes). Percentages of FGFR3 mutations, and amplifications of the FGF3, FGF4 and FGF19 genes in the consensus molecular classes are presented in stack plots. Effect sizes and p values are displayed in the plot captions. Numbers of observations in the consensus classes are indicated in the X axes._
_LumP: luminal papillary; LumU: luminal genetically unstable; LumNS: luminal non-specified; Stroma-rich: luminal stroma-rich; Ba/Sq: basal/squamous-like; NE-like: neuroendocrine-like._

\newpage

```{r tab-cons-dge, tab.cap = "Expression of FGF- and FGFR-coding genes in the consensus molecular classes of urothelial cancers. log2-transformed expression levels are presented as medians with interquartile ranges and ranges. Statistical significance of differences between the molecular classes was determined by one-way ANOVA with eta-square effect size statistic. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file."}

rep_tabs$cons_dge[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r fig-sub-fgfr-expression, fig.width = figur::convert(rep_figs$sub_fgfr_expression, to = 'in')$w, fig.height = figur::convert(rep_figs$sub_fgfr_expression, to = 'in')$h, fig.cap = 'Differential expression of FGFR-coding genes in the consensus molecular classes of urothelial cancers.'}

rep_figs$sub_fgfr_expression$plot

```

__Figure \@ref(fig:fig-sub-fgfr-expression). Differential expression of FGFR-coding genes in the consensus molecular classes of urothelial cancers.__ 

_Assignment of cancer samples from the `r globals$cohort_labs[c("tcga", "imvigor")] %>% paste(collapse = ' and ')` to the consensus molecular classes was presented in Figure \@ref(fig:fig-fgfr-subtypes). Differences in log2-transformed mRNA levels of FGF- and FGFR-coding genes between the consensus molecular classes were assessed by one-way ANOVA with $\eta^2$ effect size statistic. P values were corrected for multiple testing with the false discovery rate method._
_Expression of FGFR1 and FGFR3 was found to differ significantly between the consensus classes in both cohorts. Median log2-transformed mRNA levels with interquartile ranges are depicted as boxes with whiskers spanning over 150% of the interquartile ranges. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of cancer samples in the consensus classes are indicated in the X axes._ 
_LumP: luminal papillary; LumU: luminal genetically unstable; LumNS: luminal non-specified; Stroma-rich: luminal stroma-rich; Ba/Sq: basal/squamous-like; NE-like: neuroendocrine-like._

\newpage

```{r fig-sub-fgf-expression1, fig.width = figur::convert(rep_figs$sub_fgf_expression1, to = 'in')$w, fig.height = figur::convert(rep_figs$sub_fgf_expression1, to = 'in')$h, fig.cap = 'Differential expression of FGF-coding genes in the consensus molecular classes of urothelial cancers.'}

rep_figs$sub_fgf_expression1$plot

```

__Figure \@ref(fig:fig-sub-fgf-expression1). Differential expression of FGF-coding genes in the consensus molecular classes of urothelial cancers.__ 

_Assignment of cancer samples from the `r globals$cohort_labs[c("tcga", "imvigor")] %>% paste(collapse = ' and ')` to the consensus molecular classes was presented in Figure \@ref(fig:fig-fgfr-subtypes). Differences in log2-transformed mRNA levels of FGF- and FGFR-coding genes between the consensus molecular classes were assessed by one-way ANOVA with $\eta^2$ effect size statistic. P values were corrected for multiple testing with the false discovery rate method._
_Expression of FGF2, FGF5, and FGF7 was found to differ significantly between the consensus classes in both cohorts. Median log2-transformed mRNA levels with interquartile ranges are depicted as boxes with whiskers spanning over 150% of the interquartile ranges. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of cancer samples in the consensus classes are indicated in the X axes._ 
_LumP: luminal papillary; LumU: luminal genetically unstable; LumNS: luminal non-specified; Stroma-rich: luminal stroma-rich; Ba/Sq: basal/squamous-like; NE-like: neuroendocrine-like._

\newpage

```{r fig-sub-fgf-expression2, fig.width = figur::convert(rep_figs$sub_fgf_expression2, to = 'in')$w, fig.height = figur::convert(rep_figs$sub_fgf_expression2, to = 'in')$h, fig.cap = 'Differential expression of FGF-coding genes in the consensus molecular classes of urothelial cancers.'}

rep_figs$sub_fgf_expression2$plot

```

__Figure \@ref(fig:fig-sub-fgf-expression2). Differential expression of FGF-coding genes in the consensus molecular classes of urothelial cancers.__ 

_Assignment of cancer samples from the `r globals$cohort_labs[c("tcga", "imvigor")] %>% paste(collapse = ' and ')` to the consensus molecular classes was presented in Figure \@ref(fig:fig-fgfr-subtypes). Differences in log2-transformed mRNA levels of FGF- and FGFR-coding genes between the consensus molecular classes were assessed by one-way ANOVA with $\eta^2$ effect size statistic. P values were corrected for multiple testing with the false discovery rate method._
_Expression of FGF10 and FGF18 was found to differ significantly between the consensus classes in both cohorts. Median log2-transformed mRNA levels with interquartile ranges are depicted as boxes with whiskers spanning over 150% of the interquartile ranges. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of cancer samples in the consensus classes are indicated in the X axes._ 
_LumP: luminal papillary; LumU: luminal genetically unstable; LumNS: luminal non-specified; Stroma-rich: luminal stroma-rich; Ba/Sq: basal/squamous-like; NE-like: neuroendocrine-like._

\newpage

## Genetic subsets of urothelial carcinoma

```{r fig-lca-devel, fig.width = figur::convert(rep_figs$lca_devel, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_devel, to = 'in')$h, fig.cap = 'Development of genetic subsets of urothelial carcinoma in the GENIE BLCA training cohort. Prediction of the genetic subset assignment for the TCGA BLCA samples.'}

rep_figs$lca_devel$plot

```

__Figure \@ref(fig:fig-lca-devel). Development of genetic subsets of urothelial carcinoma in the GENIE BLCA training cohort. Prediction of the genetic subset assignment for the TCGA BLCA samples.__ 

_The genetic subsets of urothelial cancers were developed in the `r globals$cohort_labs["genie"]` training cohort by latent class modeling with binary indexes (presence/absence) of genetic alterations present in at least 5% of samples as modeling factors. Presence of FGF3, FGF4, FGF19, or CCND1 amplification was subsumed under 11q13 amplification prior to the modeling. Presence of CDKN2A, CDKN2B, or MTAP deletions was subsumed under 9p21 deletion prior to the modeling._

_(A) Selection of the genetic subset number in the `r globals$cohort_labs["genie"]` training cohort based on log-likelihood and Bayesian Information Criterion (BIC). Note the minimal BIC for the 6-subset model, which is subsequently used for definition of the genetic subsets._

_(B) The 6-subset model was trained in the `r globals$cohort_labs['genie']` cohort and used for assignment of cancer samples from the `r globals$cohort_labs["tcga"]` collective to the genetic subsets. Class assignment probabilities (posterior p) of single cancer samples are visualized in scatter plots. Note that high posterior p indicate high confidence at class assignment and stability of the modeling solution. Total observation numbers are displayed in the plot captions. Numbers of samples in the genetic subsets are indicated in the plot facets._ 

\newpage

```{r fig-lca-size, fig.width = figur::convert(rep_figs$lca_size, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_size, to = 'in')$h, fig.cap = 'Distribution of sizes of the genetic subsets in the training GENIE BLCA cohort and the test TCGA BLCA collective.'}

rep_figs$lca_size$plot

```

__Figure \@ref(fig:fig-lca-size). Distribution of sizes of the genetic subsets in the training GENIE BLCA cohort and the test TCGA BLCA collective.__ 

_Genetic subsets were developed in the `r globals$cohort_labs["genie"]` cohort and predicted for cancers samples in the `r globals$cohort_labs["tcga"]` as presented in Figure \@ref(fig:fig-lca-devel). Sizes of the genetic subsets expressed as percentages of cancer samples in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` collectives are presented in a stack plot._

\newpage

```{r tab-lca-genetics, tab.cap = "Frequency of the most frequent somatic mutations and copy number alterations in the genetic subsets of urothelial cancers. The frequencies are presented as percentages of samples with genetic alterations in the genetic subsets. Differences between the genetic subsets were assessed by chi-squared test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file."}

rep_tabs$lca_genetics[1, 1] %>% 
  mutate(Cohort = '') %>% 
  set_names(' ') %>% 
  flextable %>% 
  set_widths(16)

```

\newpage

```{r fig-lca-genet, fig.width = figur::convert(rep_figs$lca_genet, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_genet, to = 'in')$h, fig.cap = 'Distribution of the most frequent somatic mutations and copy number alterations in the genetic subsets of urothelial cancers.'}

rep_figs$lca_genet$plot

```

__Figure \@ref(fig:fig-lca-genet). Distribution of the most frequent somatic mutations and copy number alterations in the genetic subsets of urothelial cancers.__ 

_Differences in frequency of the most common somatic mutations and copy number variants ($\geq$ 5% of samples) between the genetic subsets of urothelial cancers were assessed by $\chi^2$ test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Presence of the genetic features in the genetic subsets was presented in a heat map. Significant alterations are highlighted in the Y axis with bold font. Numbers of analyzed cancer samples in the subsets are displayed in the plot captions._

\newpage

```{r fig-lca-details-mutations, fig.width = figur::convert(rep_figs$lca_details_mutations, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_details_mutations, to = 'in')$h, fig.cap = 'Frequency of selected somatic mutations in the genetic subsets of urothelial cancers.'}

rep_figs$lca_details_mutations$plot

```

__Figure \@ref(fig:fig-lca-details-mutations). Frequency of selected somatic mutations in the genetic subsets of urothelial cancers.__ 

_Differences in frequency of TP53, RB1, FGFR3, and ERBB2 mutations between the genetic subsets of urothelial cancers were assessed by $\chi^2$ test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Percentages of the mutations in the genetic subsets are presented in stack plots. Effect sizes and p values are displayed in the plot captions. Numbers of cancer samples in the subsets are indicated in the X axes._

\newpage

```{r fig-lca-details-cna, fig.width = figur::convert(rep_figs$lca_details_cna, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_details_cna, to = 'in')$h, fig.cap = 'Frequency of selected copy number alterations in the genetic subsets of urothelial cancers.'}

rep_figs$lca_details_cna$plot

```

__Figure \@ref(fig:fig-lca-details-cna). Frequency of selected copy number alterations in the genetic subsets of urothelial cancers.__ 

_Differences in frequency of CCND1 amplifications (hallmark alteration for 11q13 chromosome region amplification), E2F3 amplifications, MDM2 amplifications, and CDKN2A deletions (characteristic for 9p21 chromosome region deletion) between the genetic subsets of urothelial cancers were assessed by $\chi^2$ test with Cramer's V effect size statistic. P values were corrected for multiple testing with the false discovery rate method. Percentages of the copy number variants in the genetic subsets are presented in stack plots. Effect sizes and p values are displayed in the plot captions. Numbers of cancer samples in the subsets are indicated in the X axes._

\newpage

```{r fig-lca-tmb, fig.width = figur::convert(rep_figs$lca_tmb, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_tmb, to = 'in')$h, fig.cap = 'Total mutation burden, and counts of somatic mutations and copy number alterations in the genetic subsets of urothelial cancers.'}

rep_figs$lca_tmb$plot

```

__Figure \@ref(fig:fig-lca-tmb). Total mutation burden, and counts of somatic mutations and copy number alterations in the genetic subsets of urothelial cancers.__ 

_Total mutation burdens (TMB) were provided for the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` cohorts by the study authors. Counts of mutations, deletions and amplifications were computed for each cancer sample based on the published information. TMB, and counts of mutations, deletions and amplifications were compared between the genetic subsets of urothelial cancers by Kruskal-Wallis test with $\eta^2$ effect size statistic. P values were adjusted for multiple testing with the false discovery rate method. Median values with interquartile ranges are shown as boxes with whiskers spanning over 150% of the interquartile ranges. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of analyzed cancer samples are indicated in the X axes._

\newpage

```{r tab-lca-dge, tab.cap = 'Expression of FGF- and FGFR-coding genes in the genetic subsets of urothelial cancers of the TCGA BLCA cohort. log2-transformed expression levels are presented as medians with interquartile ranges and ranges. Statistical significance of differences between the genetic subsets was determined by one-way ANOVA with eta-square effect size statistic. P values were corrected for multiple testing with the false discovery rate method. The table is available as a supplementary Excel file.'}

rep_tabs$lca_dge[1, 1] %>% 
  mutate(Variable = '') %>% 
  set_names(' ') %>% 
  flextable %>%
  set_widths(16)

```

\newpage

```{r fig-lca-dge-fgfr, fig.width = figur::convert(rep_figs$lca_dge_fgfr, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_dge_fgfr, to = 'in')$h, fig.cap = 'Differential expression of FGFR-coding genes in the genetic subsets of urothelial cancers.'}

rep_figs$lca_dge_fgfr$plot

```

__Figure \@ref(fig:fig-lca-dge-fgfr). Differential expression of FGFR-coding genes in the genetic subsets of urothelial cancers.__ 

_Differences in log2-transformed mRNA levels of FGF- and FGFR-coding genes were compared between the genetic subsets of urothelial cancers in the `r globals$cohort_labs["tcga"]` cohort by one-way ANOVA with $\eta^2$ effect size statistic. Differences in expression of FGFR1, FGFR2, and FGFR3 were found significant. Median log2-transformed expression values with interquartile ranges are depicted as boxes with whiskers spanning over 150% of the interquartile ranges. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of analyzed cancer samples are indicated in the X axes._

\newpage

```{r fig-lca-dge-fgf, fig.width = figur::convert(rep_figs$lca_dge_fgf, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_dge_fgf, to = 'in')$h, fig.cap = 'Differential expression of FGF-coding genes in the genetic subsets of urothelial cancers.'}

rep_figs$lca_dge_fgf$plot

```

__Figure \@ref(fig:fig-lca-dge-fgf). Differential expression of FGF-coding genes in the genetic subsets of urothelial cancers.__ 

_Differences in log2-transformed mRNA levels of FGF- and FGFR-coding genes were compared between the genetic subsets of urothelial cancers in the `r globals$cohort_labs["tcga"]` cohort by one-way ANOVA with $\eta^2$ effect size statistic. Differences in expression of FGF2, FGF5, FGF7, and FGF17 were found significant. Median log2-transformed expression values with interquartile ranges are depicted as boxes with whiskers spanning over 150% of the interquartile ranges. Single cancer samples are visualized as points. Effect sizes and p values are displayed in the plot captions. Numbers of analyzed cancer samples are indicated in the X axes._

\newpage

```{r tab-lca-clinic, tab.cap = 'Demographic, clinical, and pathological characteristic of the genetic subsets of urothelial cancers. Numeric features are presented as medians with interquartile ranges and ranges. Qualitative variables are shown as percentages and observation counts of the categories.'}

flextable::flextable(rep_tabs$lca_clinic) %>% 
  set_widths(c(1.5, 2, rep(2.1, 6), 2.3, 2.1)) %>% 
  merge_v(1) %>%
  footnote(1, 9:10, 
           value = as_paragraph('Numeric variables: Kruskal-Wallis test with \u03B7\u00B2 effect size statistic. Categorical variables: \u03C7\u00B2 test with Cramer V effect size statistic.'), 
           part = 'header', 
           ref_symbols = 'a') %>% 
  theme_vanilla

```

\newpage

```{r fig-lca-clinic, fig.width = figur::convert(rep_figs$lca_clinic, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_clinic, to = 'in')$h, fig.cap = 'Age and gender in the genetic subsets.'}

rep_figs$lca_clinic$plot

```

__Figure \@ref(fig:fig-lca-clinic). Age and gender in the genetic subsets.__ 

_Differences in age and gender between the genetic subsets of urothelial cancers in the `r globals$cohort_labs[c("genie", "tcga")] %>% paste(collapse = ' and ')` were investigated by Kruskal-Wallis test and $\chi^2$ test, respectively. $\eta^2$ and Cramer's V served as effect size metrics. P values were corrected for multiple testing with the false discovery rate method. Median age with the interquartile range is presented in a box plot with whiskers spanning over 150% of the interquartile range and single cancer patients visualized as points. Percentages of women and men in the genetic subsets are shown in stack plots. Effect sizes and p values are displayed in the plot captions. Numbers of observations in the genetic subsets are indicated in the X axes._

\newpage

```{r fig-lca-os, fig.width = figur::convert(rep_figs$lca_os, to = 'in')$w, fig.height = figur::convert(rep_figs$lca_os, to = 'in')$h, fig.cap = 'Overall survival in the genetic subsets of urothelial cancers.'}

rep_figs$lca_os$plot

```

__Figure \@ref(fig:fig-lca-os). Overall survival in the genetic subsets of urothelial cancers.__ 

_Overall survival information was provided for the `r globals$cohort_labs["tcga"]` study participants. Approximate overall survival was calculated for the `r globals$cohort_labs["genie"]` cohort based on age at enrollment in years and age at the last follow-up or death in days. Overall survival was subsequently compared between the genetic subsets by false discovery rate-corrected Peto-Peto test. Fractions of surviving patients are presented in Kaplan-Meier plots. P values for globals differences in survival between the subsets are indicated in the plots, numbers of observations and deaths are displayed in the plot captions. Numbers of observations and deaths in the FGFR3 mutation strata are presented in the plot legends._

\newpage

# Data and code availability

Data from publicly available sources were analyzed. 
The R analysis pipeline is available as a [GitHub repository](https://github.com/PiotrTymoszuk/FGFR-BLCA)

\newpage

# References
